### PR TITLE
Peripherals 2

### PR DESCRIPTION
**Pull request by kgliwinski**
- TMC2209 to be fixed later
- Distance sensor driver will be done separately

### CODE REVIEW

File: src/components/peripheral_drivers/adc_driver/adc_driver.c [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+adc_driver_status_t adc_driver_convert_reading(adc_driver_t *config,
+                                               uint8_t *result,
+                                               uint32_t *converted_result,
+                                               uint32_t *converted_channel) {
+  if (!config->initialized) {
+    return ADC_DRIVER_ERROR;
+  }
+
+  adc_digi_output_data_t *p = (adc_digi_output_data_t *)result;
```
Comment (JSiuda1) [inline src/components/peripheral_drivers/adc_driver/adc_driver.c:100]: Looks sketchy

File: src/components/peripheral_interfaces/adc_wrapper/adc_wrapper.c [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+
+  uint32_t actual_readings = 0;
+  uint32_t channel = 0;
+  while (1) {
+    adc_driver_status_t status =
+        adc_driver_get_reading(&adc_driver, buf, READINGS_NUM, &out_length, 0);
+
+    if (status == ADC_DRIVER_OK) {
+      for (uint32_t i = 0; i < out_length; i += SOC_ADC_DIGI_RESULT_BYTES) {
+        adc_driver_convert_reading(&adc_driver, &buf[i], &readings_out[i],
```
Comment (JSiuda1) [inline src/components/peripheral_interfaces/adc_wrapper/adc_wrapper.c:78]: Together with this increase, it is hard for me to say what is going on here without reading IDF ADC documents.

File: src/components/app/sensor_task/sensor_task.c
Comment (JSiuda1) [inline src/components/app/sensor_task/sensor_task.c:1]: I am a bit confused about the GPIO and ADC, but if I understand this correctly, we only read the measurements from the load cell here?

File: src/components/ic_drivers/test/shared_gpio_wrapper.c
Comment (JSiuda1) [inline src/components/ic_drivers/test/shared_gpio_wrapper.c:1]: I see there you want to use on gpio class to control the expander gpios and esp gpios? Am I correct? I am trying to check if I understand what is going on here.

**Review by JSiuda1 — APPROVED** [review]
It's too big to check everything out in a reasonable amount of time, but in overall it looks OK.🍡

File: src/components/ic_drivers/tmc2209_driver/tmc2209_driver.c [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+
+  tmc2209_driver_status_t ret = ((uint8_t)ifcnt.raw == if_cnt + 2)
+                                    ? TMC2209_DRIVER_OK
+                                    : TMC2209_DRIVER_ERROR;
+  if (ret != TMC2209_DRIVER_OK) {
+    return ret;
+  }
+  return TMC2209_DRIVER_OK;
+}
+
```
Comment (JSiuda1) [inline src/components/ic_drivers/tmc2209_driver/tmc2209_driver.c:81]: Can you add set veliocity to 0 before enabling the driver, this should fix the problem with mottor running after esp reset.

File: src/components/peripheral_drivers/adc_driver/adc_driver.c (hunk 1 above)
Comment (kgliwinski) [reply src/components/peripheral_drivers/adc_driver/adc_driver.c:100]: This is getting refactored anyways but yeah, that's how they do it in the example

File: src/components/ic_drivers/test/shared_gpio_wrapper.c
Comment (kgliwinski) [reply src/components/ic_drivers/test/shared_gpio_wrapper.c:1]: This is the part for testing and was done in a sort of hurry when I realised I can't comfortably open/close solenoids. Anyhow planning to change these a bit later

File: src/components/app/sensor_task/sensor_task.c
Comment (kgliwinski) [reply src/components/app/sensor_task/sensor_task.c:1]: Yeah, ADC is only for loadcell (at least in the current v1 board configuration) and will be changed a bit

File: src/components/ic_drivers/tmc2209_driver/tmc2209_driver.c (hunk 3 above)
Comment (kgliwinski) [reply src/components/ic_drivers/tmc2209_driver/tmc2209_driver.c:81]: Added but as this driver is still non-functional will be uncommented later

**Pull request merged**